main.ts

```diff
@@ -39,16 +49,23 @@ function sleep(ms: number) {
 	return new Promise(resolve => setTimeout(resolve, ms));
 }
 
+class YankEvent extends Events {
+	on(name: 'vim-yank', callback: (text: string) => void): EventRef;
+	on(name: 'vim-change', callback: (text: string) => void): EventRef;
+	on(name: 'vim-delete', callback: (text: string) => void): EventRef;

```

Comment:
Just a thought for maintainers that it may be useful to know where these event names come from? I'd add that in a comment.

Reply:
I actually didn't impelement that well, I fixed that now.

Reply:
I was approaching it more from the perspective of documentation. I, for one, have no idea where the event names come from, so a comment saying that "these event names come from <here>" would be great imo.

Reply:
I made them up. They're events I created myself and which are getting triggered by the patched method.

Reply:
_OH_. I'm a moron!

Reply:
No worries! :smile: